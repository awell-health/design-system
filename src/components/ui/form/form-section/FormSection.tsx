import { FC } from 'react';

interface Props {
  title: string;
  hint?: string;
  children: React.ReactNode | JSX.Element | string;
}

export const FormSection: FC<Props> = ({ children, title, hint = null }) => {
  return (
    <div className='py-6 flex flex-col gap-2'>
      <div className='flex flex-col gap-1'>
        <h5 className='text-lg font-medium leading-5 text-neutral-dark-800'>{title}</h5>
        {hint !== null && <div className='text-gray-500'>{hint}</div>}
      </div>
      <div className='mt-2'>{children}</div>
    </div>
  );
};
