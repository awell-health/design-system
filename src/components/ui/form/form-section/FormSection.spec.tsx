import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { FormSection } from './FormSection';

describe('FormSection', () => {
  const subject = (props = {}) => (
    <FormSection title='Test Title' {...props}>
      <div>Test Content</div>
    </FormSection>
  );

  it('match snapshot', () => {
    const { container } = render(subject());
    expect(container).toMatchSnapshot();
  });

  it('renders the title', () => {
    render(subject());
    expect(screen.getByText('Test Title')).toBeInTheDocument();
  });

  it('renders the hint when provided', () => {
    render(subject({ hint: 'Test Hint' }));
    expect(screen.getByText('Test Hint')).toBeInTheDocument();
  });

  it('does not render the hint when not provided', () => {
    render(subject());
    expect(screen.queryByText('Test Hint')).not.toBeInTheDocument();
  });

  it('renders children content', () => {
    render(subject({ children: 'Test Content' }));
    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });

  it('renders the required asterisk when isRequired is true', () => {
    render(subject({ isRequired: true }));
    expect(screen.getByText('*')).toBeInTheDocument();
  });

  it('does not render the required asterisk when isRequired is false', () => {
    render(subject());
    expect(screen.queryByText('*')).not.toBeInTheDocument();
  });

  it('renders the separator when showSeparator is true', () => {
    render(subject({ showSeparator: true }));
    expect(screen.getByRole('separator')).toBeInTheDocument();
  });

  it('does not render the separator when showSeparator is false', () => {
    render(subject({ showSeparator: false }));
    expect(screen.queryByRole('separator')).not.toBeInTheDocument();
  });
});
