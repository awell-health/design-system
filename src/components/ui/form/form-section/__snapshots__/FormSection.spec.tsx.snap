// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormSection > match snapshot 1`] = `
<div>
  <div
    class="py-4 flex flex-col gap-2"
  >
    <div
      class="flex flex-col"
    >
      <h5
        class="text-lg font-medium leading-5 text-gray-600"
      >
        Test Title
      </h5>
    </div>
    <div>
      Test Content
    </div>
  </div>
</div>
`;
